internal/v2/controller/http/controller.go

```diff
@@ -0,0 +1,80 @@
+//
+// Copyright (c) 2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package http
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+	commonV2 "github.com/edgexfoundry/go-mod-core-contracts/v2"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/common"
+	"github.com/google/uuid"
+)
+
+// V2Controller controller for V2 REST APIs
+type V2Controller struct {
+	lc logger.LoggingClient
+}
+
+// NewV2Controller creates and initializes an V2Controller
+func NewV2Controller(lc logger.LoggingClient) *V2Controller {
+	return &V2Controller{
+		lc: lc,
+	}
+}
+
+// NewPingResponse
+// TODO: move into go-mod-core-contracts
+func NewPingResponse() common.PingResponse {
+	return common.PingResponse{
+		// TODO: add apiversion
+		Timestamp: time.Now().Format(time.UnixDate),
+	}
+}
+
+// Ping
+func (v2c *V2Controller) Ping(w http.ResponseWriter, r *http.Request) {
+	pingResponse := NewPingResponse()
+
+	v2c.SendResponse(w, commonV2.ApiPingRoute, pingResponse, uuid.New().String())
+
+}
+
+// SendResponse
+func (v2c *V2Controller) SendResponse(w http.ResponseWriter, api string, item interface{}, correlationID string) {
+	data, err := json.Marshal(item)
+	if err != nil {
+		v2c.lc.Error(fmt.Sprintf("Unable to marshal %s response", api), "error", err.Error(), clients.CorrelationHeader, correlationID)
```

Comment:
rewording; error message kind of confusion: are we marshalling `api` or `item`?

Reply:
`api` is passed in and contains the API path, e.g. `/api/v2/ping`